python/hail/expr/expressions/typed_expressions.py

```diff
@@ -1309,7 +1309,7 @@ def select(self, *fields, **named_exprs):
 
         indices, aggregations = unify_all(self, kwargs_struct)
 
-        return construct_expr(ApplyMethod('merge', StructOp('select', self._ast, *select_names), kwargs_struct._ast),
+        return construct_expr(ApplyMethod('annotate', StructOp('select', self._ast, *select_names), kwargs_struct._ast),
```

Comment:
@tpoterba doctests caught this usage of `merge` in python, but since field collisions are being checked already I just replaced it with `annotate`.